File: Java/CodeWars/src/SumsOfParts.java
public class SumsOfParts {

    public static int[] Sum(int[] ls){
        int[] res = new int[ls.length+1];
        for (int i = ls.length-1; i >= 0; i--){
            res[i] = res[i+1] + ls[i];
        }
        return res;
    }
    public static void main(String[] args){
        int[] ls = {1,2,3,4,5,6}, res = Sum(ls);
        for (int v:res){
            System.out.print(v+", ");
        }
    }
}

File: Javascript/CodeWars/Sums of Parts.js
function partsSums(ls) {
    var res = [ls.reduce((tot,x)=>tot+=x,0)];
    for(let x of ls){
        res.push(res[res.length-1]-x);
    } 
    return res;
}
ls = [1,2,3,4,5,6]
console.log(partsSums(ls));